### PR TITLE
Enable read keyword in backup and restore

### PR DESCRIPTION
**Pull request by branupama**
This commit adds the changes to enable read keyword’s value
from DBus in the backup & restore functionality.

### CODE REVIEW

File: src/backup_restore.cpp [hunk 1]
```diff
@@ -157,6 +157,33 @@ void BackupAndRestore::backupAndRestoreIpzVpd(types::IPZVpdMap& io_srcVpdMap,
...
+    if (l_srcFruPath.empty() || l_dstFruPath.empty())
+    {
+        logging::logMessage(
+            "Couldn't find either source or destination FRU path.");
+        return;
+    }
+
+    const std::string& l_srcServiceName =
+        m_sysCfgJsonObj["frus"][l_srcFruPath][0]["serviceName"];
+    const std::string& l_dstServiceName =
```
Comment (SunnySrivastava1984) [inline src/backup_restore.cpp:173]: Have a json utility function to get service name?

**Review by SunnySrivastava1984 — CHANGES_REQUESTED** [review]
As per the commit message I don't see any read Keyword's API's getting enabled.

File: src/backup_restore.cpp (hunk 1 above)
Comment (branupama) [reply src/backup_restore.cpp:173]: added new utility API to get service name.

**Comment by branupama** [comment]
> As per the commit message I don't see any read Keyword's API's getting enabled.

updated the commit message.

**Review by jinuthomas — CHANGES_REQUESTED** [review]
using json utility functions in code will throw and neither you are catching it here , and you are letting it pass and the oxygen of the function does not tell me that it will throw. and has to be handled by the caller. Please fix that. Either catch it here and log and error or something else as I cannot track where the error I logged on this failure to do backup and restore for IPZ vpd

**Comment by branupama** [comment]
> using json utility functions in code will throw and neither you are catching it here , and you are letting it pass and the oxygen of the function does not tell me that it will throw. and has to be handled by the caller. Please fix that. Either catch it here and log and error or something else as I cannot track where the error I logged on this failure to do backup and restore for IPZ vpd

Updated doxygen for backupAndRestoreIpzVpd() API can throw an error.

File: src/backup_restore.cpp [hunk 2]
```diff
@@ -157,6 +157,39 @@ void BackupAndRestore::backupAndRestoreIpzVpd(types::IPZVpdMap& io_srcVpdMap,
         return;
     }
 
+    const std::string l_srcFruPath =
```
[the remark this answers is not in the document]
Comment (branupama) [reply src/backup_restore.cpp]: It is needed for updating the keyword's value on hardware.
for now its not being used, so removed these variable.

File: src/backup_restore.cpp [hunk 3]
```diff
@@ -157,6 +157,39 @@ void BackupAndRestore::backupAndRestoreIpzVpd(types::IPZVpdMap& io_srcVpdMap,
...
+        jsonUtility::getFruPathFromJson(m_sysCfgJsonObj, i_srcPath);
+    const std::string l_dstFruPath =
```
[the remark this answers is not in the document]
Comment (branupama) [reply src/backup_restore.cpp]: It is needed for updating the keyword's value on hardware.
for now its not being used, so removed these variable.

File: src/backup_restore.cpp [hunk 4]
```diff
@@ -157,6 +157,39 @@ void BackupAndRestore::backupAndRestoreIpzVpd(types::IPZVpdMap& io_srcVpdMap,
...
+    const std::string l_dstInvPath =
+        jsonUtility::getInventoryObjPathFromJson(m_sysCfgJsonObj, i_dstPath);
+    if (l_srcInvPath.empty() || l_dstInvPath.empty())
+    {
+        logging::logMessage(
+            "Couldn't find either source or destination inventory path.");
+        return;
+    }
+
+    const std::string l_srcServiceName =
```
[the remark this answers is not in the document]
Comment (branupama) [reply src/backup_restore.cpp:171]: as l_srcServiceName is used in the loop. So getting this service name at the  beginning only.

File: src/backup_restore.cpp [hunk 5]
```diff
@@ -157,6 +157,39 @@ void BackupAndRestore::backupAndRestoreIpzVpd(types::IPZVpdMap& io_srcVpdMap,
...
+        jsonUtility::getServiceName(m_sysCfgJsonObj, l_srcInvPath);
+    const std::string l_dstServiceName =
```
[the remark this answers is not in the document]
Comment (branupama) [reply src/backup_restore.cpp:173]: as l_dstServiceName is used in the loop. So getting this service name at the  beginning only.

**Review by SunnySrivastava1984 — APPROVED** [review]

**Review by jinuthomas — APPROVED** [review]

**Pull request merged**